languages/go/exercises/concept/using-map/.docs/after.md

```diff
@@ -0,0 +1,35 @@
+`map` in go is a key-value type. In other programming language, you might know map as `dict` or associative array. The value of an uninitialized map is `nil`, and you can define map as follows
+
+`make(map[string]int)`
+
+or
+
+```go
+  var foo map[string]int
+  foo = make(map[string]int)
+```
+
+> Here `string` represent the key type of map, and `int` represent the element type of map. You can change them to whatever valid golang type.
+
+To store a value in a map, you can use `=` operator like
+
+`foo["bar"] = 42`
+
+`map` is indexed by its key, therefore, you are required to use different key if you want to store a different data.
+
+To get value from a map, you can use
+
+`foo["bar"]`
+
+Retrieving inexistent key from map will cause a runtime error, and to check if such a key exists you can do
+
+`value, exists := foo["baz"]`
+
+To delete an item from a map, you can use
+
+`delete(foo, "bar")`
+
+If you try to write to a `map` from multiple go routine, that trigger the race detector, [see this link][godoc-race-detector] and [here][goblog-race-detector]. Alternatively, you can use `sync.Map` or `atomic` or `mutex` to work around this issue.
```

Comment:
This is a great bit of information to add to the after document!

```suggestion
If you try to write to a `map` from multiple goroutines, that trigger the race detector, [see this link][godoc-race-detector] and [here][goblog-race-detector]. Alternatively, you can use `sync.Map` or `atomic` or `mutex` to work around this issue.
```

Reply:
Yes definitely very important information!

Not sure if it is too much for this document, but sync.Map is not meant as a replacement for maps + mutex. sync.Map is good when reading a lot from the map especially in loops but is not very performant when there are many writes. The usual way to use a map in a concurrent environment is with mutexes.

Reply:
Maybe we can dumb it down a bit and state something a bit more generically, like how using maps in concurrent scenarios can be problematic? I would indeed prefer not naming any of the solutions.